src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/model/CaseFieldPathUtils.java

```diff
@@ -79,6 +80,15 @@ private CaseFieldPathUtils() {
                       pathIncludesParent ? getPathElementsTail(pathElements) : pathElements);
     }
 
+    public static JsonNode getNestedCaseFieldByPath(Map<String, JsonNode> caseData, String path) {
+        List<String> pathElements = getPathElements(path);
+        JsonNode topLevelNode = caseData.get(pathElements.get(0));
+
+        return pathElements.size() > 1 && topLevelNode != null && !topLevelNode.isNull()
```

Comment:
Are both null checks on topLevelNode required? I guess the JsonNode isNull() checks if this node was created from JSON literal null value but just wondered if the two checks were necessary?

Reply:
Yep, `isNull()` checks whether it's a `NullNode` (if the JSON key exists but value is null), whereas the other null check is relevant if the key doesn't exist 👍